src/task/screen-bad-words-task.ts

```diff
@@ -0,0 +1,51 @@
+// global
+import { DatabaseTransactionHandler, Member } from 'graasp';
+import BadWordsFilter from 'bad-words';
+import { stripHtml } from '../utils';
+// local
+import { ValidationService } from '../db-service';
+import { BaseValidationTask } from '../task/base-validation-task';
+
+type InputType = { itemId: string };
+
+export class ScreenBadWordsTask extends BaseValidationTask<boolean> {
+  input: InputType;
+
+  get name(): string {
+    return ScreenBadWordsTask.name;
+  }
+
+  constructor(member: Member, validationService: ValidationService, input: InputType) {
+    super(member, validationService);
+    this.input = input;
+  }
+
+  checkBadWrods = (documents: string[]) => {
+    const strings = documents?.filter(Boolean).map((entry) => stripHtml(entry));
+    const badWordsFilter = new BadWordsFilter();
+    for (const index in strings) {
+      if (badWordsFilter.isProfane(strings[index])) {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  async run(handler: DatabaseTransactionHandler): Promise<void> {
+    this.status = 'RUNNING';
+
+    const { itemId } = this.input;
+    const item = await this.validationService.getItem(itemId, handler);
+    const suspicious = this.checkBadWrods([item.name, item.description]);
+    if (suspicious) {
+      this.status = 'FAIL';
+      this._result = true;
```

Comment:
What about sending back words that causes the check to fail?

Reply:
With the current library, I don't think it's feasible to tell exactly which words fail the check, because the library only provides two types: either return a Boolean if the given string is suspicious, or replace bad words with marks like ***. However I could tell the user if the title or the description is suspicious.
The other way is write the checking algorithm from scratch, but I think this way would be error-prone, and it's better to use state-of-arts libraries, which are thoroughly tested with guaranteed performance. I think I could also take a look for a better suited library, although so far this one is the most suitable one I found.

Reply:
I see, then we can maybe do as you suggest, return which field is suspicious.